contributors/design-proposals/testing/trigger-pr-tests.md

```diff
@@ -0,0 +1,92 @@
+
+@cjwagner
+> We could check `review_comment` events and remove the `do-not-merge/needs-ok-to-test` label if the comment contains `/ok-to-test`, but we would still need to list all reviews and review comments when checking if a non-member PR is trusted to prevent race conditions with other events.
+
+> Using a positive label instead of a negative one avoids all the race conditions that force us to list comments and check for `/ok-to-test` instead of just checking for a label.
+
+## Proposal
+
+We suggest introducing a new label named `ok-to-test` that would be required on any non-member PR before automatic test jobs can be started by the `trigger` plugin.
+
+This label could only be added by members of the *trusted organization* for the repository.
+
+We will also deprecate the `/ok-to-test` command.
+
+## Caveats
+
+Switching from a negative to a positive label could create confusion and add clutter to the PR.
+
+@cblecker
+> .. label overload. Having a label just to say testing is okay.. bleh.
+
+@cjwagner
+> Positive labels seem worse for contributor experience. Negative process labels are less intrusive because they disappear as the user satisfies the requirements so they can be addressed individually and incrementally. In a positive label world, users need to know what labels are **not** present in order to address unsatisfied processes. Additionally, the label set becomes more cluttered as progress is made instead of less.
+
+On the other hand, we could also consider this label like a reward: the PR has earned the rights to have automatic test jobs.
+
+@BenTheElder
+> One more downside to positive labels in the short term is that they're
+likely to be accidentally removed when users with write access edit labels
+in the GitHub UI.
+
+This is why documentation and education are key here, and like @cjwagner reminds us:
+> we are already leaving a comment on the PR if the author needs `/ok-to-test` from a member, so authors are already receiving specific instructions on how to proceed
+
+We would just have to modify this comment.
+
+## Migration path
+
+1. declare `ok-to-test` in `label_sync`
+1. modify `trigger` plugin to check `ok-to-test` label as another condition for a trusted PR
+1. modify `trigger` plugin automatic comment and `tide` documentation to explain the new `ok-to-test` label requirements
+1. batch add `ok-to-test` label to non-members trusted PRs
+1. modify `trigger` plugin to **only** look for `ok-to-test` label when checking PR trust (code simplification)
+1. stop adding `needs-ok-to-test` label to untrusted PRs
+1. batch remove `needs-ok-to-test` label to untrusted PRs
+1. remove `needs-ok-to-test` from `label_sync`
+
+## Future evolutions
+
+In the future, we might decide to require the new label for all PRs, which means that organisation members will also need the `ok-to-test` label applied to their PRs before automatic testing can be triggered.
```

Comment:
Automatically add the `ok-to-test` label to PRs created by org members?

Reply:
Good point, I will add it.